src/analysisd/decoders/syscheck.c

```diff
@@ -1433,12 +1438,25 @@ static int fim_generate_alert(Eventinfo *lf, char *mode, char *event_type,
     char changed_attributes[OS_SIZE_256];
     snprintf(changed_attributes, OS_SIZE_256, "Changed attributes: %s\n", lf->fields[FIM_CHFIELDS].value);
 
+    cJSON *tmp = cJSON_Parse(lf->fields[FIM_HARD_LINKS].value);
```

Comment:
Is it possible that `cJSON_Parse` fails for a null input? I think it could: https://github.com/DaveGamble/cJSON/blob/master/cJSON.c#L1031

We should put this logic into an `if (lf->fields[FIM_HARD_LINKS].value)` block.